src/transaction/transaction.rs

```diff
@@ -466,14 +514,30 @@ impl TwoPhaseCommitter {
     }
 }
 
-impl Drop for TwoPhaseCommitter {
+impl Drop for Transaction {
     fn drop(&mut self) {
-        if !self.committed {
-            self.bg_worker.clone().spawn_ok(self.rollback().map(|res| {
-                if let Err(e) = res {
-                    warn!("Failed to rollback: {}", e);
-                }
-            }))
+        if self.status == TransactionStatus::Normal {
```

Comment:
I think we should panic here, rather than rolling back - I think it is a bit bad to do serious work in a dtor.